libraries/DateTime/contracts/DateTime.aes

```diff
@@ -114,13 +62,191 @@ contract DateTime =
+    require(from_timestamp =< to_timestamp, "parameter to_timestamp should be higher than from_timestamp")
+    let from_dt : date_time = parse_timestamp(from_timestamp)
+    let to_dt : date_time = parse_timestamp(to_timestamp) 
+    let months : int = to_dt.year * 12 + to_dt.month - from_dt.year * 12 - from_dt.month
+    months
+
+  public function diff_days(from_timestamp : int, to_timestamp : int) : int = 
+    require(from_timestamp =< to_timestamp, "parameter to_timestamp should be higher than from_timestamp")
+    let days : int = (to_timestamp - from_timestamp) / state.day_in_seconds
+    days
+
+  public function diff_hours(from_timestamp : int, to_timestamp : int) : int = 
+    require(from_timestamp =< to_timestamp, "parameter to_timestamp should be higher than from_timestamp")
+    let hours : int = (to_timestamp - from_timestamp) / state.hour_in_seconds
+    hours
+
+  public function diff_minutes(from_timestamp : int, to_timestamp : int) : int = 
+    require(from_timestamp =< to_timestamp, "parameter to_timestamp should be higher than from_timestamp")
+    let minutes : int = (to_timestamp - from_timestamp) / state.minute_in_seconds
+    minutes
+
+  public function diff_seconds(from_timestamp : int, to_timestamp : int) : int = 
+    require(from_timestamp =< to_timestamp, "parameter to_timestamp should be higher than from_timestamp")
+    let seconds : int = to_timestamp - from_timestamp
+    seconds
+
+  public function is_valid_date(year : int, month : int, day : int) : bool = 
+    if(year > 1970 && month > 0 && month =< 12)
+      let days_in_month = get_days_in_month(month, year)
+      if(day > 0 && day =< days_in_month)
+        true
+      else
+        false
+    else
+      false
+
+  public function is_valid_date_time(year : int, month : int, day : int, hour : int, minute : int, second : int) : bool = 
+    if(is_valid_date(year, month, day))
+      if(hour < 24 && minute < 60 && second < 60)
+        true
+      else 
+        false
+    else
+      false
+
+  public function is_week_day(timestamp : int) : bool = 
+    if(get_weekday(timestamp) =< 5)
+      true
+    else
+      false
+
+  public function is_week_end(timestamp : int) : bool = 
+    if(get_weekday(timestamp) >= 6)
+      true
+    else
+      false
+
+  public function get_month(timestamp : int) : int = 
+    let month : int = parse_timestamp(timestamp).month
+    month
+    
+  public function get_day(timestamp : int) : int = 
+    let day : int = parse_timestamp(timestamp).day
+    day
+
+  public function get_hour(timestamp : int) : int = 
+    rest(timestamp / 60 /60, 24)
+
+  public function get_minute(timestamp : int) : int = 
+    rest(timestamp / 60, 60)
+
+  public function get_second(timestamp : int) : int = 
+    rest(timestamp, 60)
+  
+  public function get_weekday(timestamp : int) : int = 
+    let weekday : int = parse_timestamp(timestamp).weekday
+    weekday
+
+  public function add_years(timestamp : int, years: int) : int =
```

Comment:
`TimeUnit` and one `add` function? (and the same goes for `sub`)

Reply:
@HristiyanG how is this resolved?